### DISCUSSION TITLE
Deprecation note on serviceMonitorSelector / podMonitorSelector

### DISCUSSION THREAD

**Discussion by xiu** [prometheus-operator/prometheus-operator#5299] [Q&A] [answered]
Hi,

I’m looking at https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.Prometheus and see that `serviceMonitorSelector` / `podMonitorSelector` have a deprecation mark. Looking at the history, I’m not sure the deprecation is about the attribute but rather a specific usecase that existed with them (https://github.com/prometheus-operator/prometheus-operator/commit/2c8a21e90952fec18952ddad4b7642121572db31). If that's the case, the comment might need to be clarified as I understood it as the attribute being deprecated at first.

Can anyone confirm?

**Comment by simonpasquier** [comment, marked as answer]
The deprecation note is about the fact that setting all selectors to nil means that the operator won't manage the Prometheus configuration secret and the object's owner is supposed to provide the configuration "manually".

https://github.com/prometheus-operator/prometheus-operator/blob/ea8bba89b3383a3c76cf0fe3b66260e5df3bf2f8/pkg/prometheus/operator.go#L1587-L1610

I agree that it could be worded more clearly.
